plugins/nimble/personal_access_token.go

```diff
@@ -0,0 +1,52 @@
+package nimble
+
+import (
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/importer"
+	"github.com/1Password/shell-plugins/sdk/provision"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/credname"
+	"github.com/1Password/shell-plugins/sdk/schema/fieldname"
+)
+
+func PersonalAccessToken() schema.CredentialType {
+	return schema.CredentialType{
+		Name:          credname.PersonalAccessToken,
+		DocsURL:       sdk.URL("https://github.com/nim-lang/nimble"),
+		ManagementURL: sdk.URL("https://nimble.directory/about.html"),
+		Fields: []schema.CredentialField{
+			{
+				Name:                fieldname.Token,
+				MarkdownDescription: "Token used to authenticate to GitHub.",
+				Secret:              true,
+				Composition: &schema.ValueComposition{
+					Length: 40,
+					Prefix: "ghp_",
+					Charset: schema.Charset{
+						Uppercase: true,
+						Lowercase: true,
+						Digits:    true,
+					},
+				},
+			},
+		},
+		DefaultProvisioner: provision.EnvVars(defaultEnvVarMapping),
+		Importer: importer.TryAll(
+			TryNimbleConfigFile(),
+			importer.TryEnvVarPair(defaultEnvVarMapping),
+		),
+	}
+}
+
+var defaultEnvVarMapping = map[string]sdk.FieldName{
+	"NIMBLE_GITHUB_API_TOKEN": fieldname.Token,
```

Comment:
Is there any significant difference between this token and the GitHub token? 

Looking at this description: `"Token used to authenticate to GitHub."`, I'd say that this token has exactly the structure of a Github Token, which we already have a `CredentialType` for. So it would be pretty cool to re-use that Github token definition here, while only creating a separate provisioner that would map the value of that GitHub token to the `NIMBLE_GITHUB_API_TOKEN` env var name (if Nimble only looks for that env var name and just plain `GITHUB_API_TOKEN` won't cut it).

Reply:
Nimble uses a `public_repo` GitHub token to publish packages, so that would be _awesome_ to have this use a standardised design!
[As far as I can tell](https://github.com/nim-lang/nimble/blob/master/src/nimblepkg/publish.nim#L22) a provisioner would be required as Nimble would need to have the `NIMBLE_GITHUB_API_TOKEN` defined, or a temporary file populated.

Reply:
Resolving this, as the following thread's discussion covers this as well.https://github.com/1Password/shell-plugins/pull/255#discussion_r1187260455